### ISSUE TITLE
NVDA does not read sub-folders accessed by address bar split button in File Explorer modal dialogs

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6750]
In File Explorer (in Windows 10), using TAB to get to the address bar (rather than alt+d which turns the address bar into one big edit field), it is possible to use the left and right arrows to move between the parts of the path.  EG in "\documents\work\letters\", you could right arrow through documents and work to letters, or from work you can down arrow and select the letters folder, or the bills folder, or awards, or whatever.

In a native File Explorer window, that works fine.  In a modal dialog such as a file open dialog from Word or Notepad, everything works up until you down arrow to browse folders.  At that point you get no speech, although the arrows do move through folders and you can select etc.  Physically moving the mouse over the drop down reads the item under the mouse.

Navigating around the dialog box and selecting folders from the tree and list view works fine.

I captured the log below when navigating through the "This PC" split button and moving the mouse onto "Downloads", then routing the focus to the mouse:

Input: kb(desktop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (08:55:12):
Developer info for navigator object:
name: u'Downloads'
role: ROLE_MENUITEM
states: STATE_SELECTABLE, STATE_FOCUSED
isFocusable: False
hasFocus: True
Python object: <NVDAObjects.Dynamic_ListItemWithoutColumnSupportMenuItemIAccessible object at 0x05895F90>
Python class mro: (<class 'NVDAObjects.Dynamic_ListItemWithoutColumnSupportMenuItemIAccessible'>, <class 'NVDAObjects.IAccessible.sysListView32.ListItemWithoutColumnSupport'>, <class 'NVDAObjects.IAccessible.MenuItem'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: (801, 217, 562, 41)
value: u'Downloads'
appModule: <'appModuleHandler' (appName u'notepad', process ID 2288) at address 58b2350>
appModule.productName: u'Microsoft\xae Windows\xae Operating System'
appModule.productVersion: u'10.0.15007.1000'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 983524
windowClassName: u'SysListView32'
windowControlID: 5
windowStyle: 1442857989
windowThreadID: 14772
windowText: u'This PC sibling list'
displayText: u'Downloads'
IAccessibleObject: <POINTER(IAccessible) ptr=0x8426908 at 584ba80>
IAccessibleChildID: 3
IAccessible event parameters: windowHandle=983524L, objectID=-4, childID=3L
IAccessible accName: u'Downloads'
IAccessible accRole: ROLE_SYSTEM_MENUITEM
IAccessible accState: STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_VALID (2097156)
IAccessible accDescription: exception: (-2147352573, 'Member not found.', (None, None, None, 0, None))
IAccessible accValue: None

**Comment by Qchristensen** [comment]
Just checked, it does work with Narrator.

**Comment by Adriani90** [comment]
Pressing tab and then shift+tab solves this. However, I can reproduce this issue and unfortunately there is not useful Information in the log I guess.

**Comment by Adriani90** [comment]
@Qchristensen I cannot reproduce this anymore with NVDA alpha-20097,6a573cd9 on Woindows 10 1909 Update. Can you confirm?

**Comment by lukaszgo1** [comment]
Even if it is fixed under W 10 1909 It still does not work for older versions of Windows. I've looked briefly and it seems fixable - I'll probably look into this after #11006 would  no longer apply.

**Comment by Qchristensen** [comment]
This issue is still the same for me using Windows 10 1909 build 18363.900 and NVDA Alpha alpha-20339,975d39d1 (and Office 365 version 16.0.4417.1000

**Comment by Adriani90** [comment]
@Qchristensen are you still seeing this issue on Windows 10 with NvDA 2023.1?

**Comment by Qchristensen** [comment]
Interestingly testing on Windows 11:

NVDA 2023.1.0.27913 
Windows 11 (64-bit) Version: 22H2, Build: 22621.1555 
Office 365 (64-bit) Version: 16.0.16227.20244 

I can still reproduce this:

1) Open Notepad
2) Press control+o to open the file open dialog
3) Tab to "Address"
4) Press down arrow to open "all locations".

Visually, a list of locations appears, and you can arrow through this and select the desired location, however NVDA does not read anything:

IO - speech.speech.speak (11:43:37.437) - MainThread (8936):
Speaking [LangChangeCommand ('en_GB'), 'All locations', 'split button', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (11:43:39.765) - winInputHook (23808):
Input: kb(desktop):downArrow
IO - inputCore.InputManager.executeGesture (11:43:40.441) - winInputHook (23808):
Input: kb(desktop):downArrow
IO - inputCore.InputManager.executeGesture (11:43:40.830) - winInputHook (23808):
Input: kb(desktop):downArrow

**Comment by Adriani90** [comment]
@Qchristensen I cannot reproduce this on Notepad version 11.2501.31.0 or on Windows Explorer in Windows 11 24 H2.
In my case a popup opens when I press down arrow and I can navigate the sub folders properly.
Is this still an issue in Windows 10?

**Comment by Adriani90** [comment]
Let's close this since we didn't hear anything from Windows 10 users on this issue since many years, and Windows 10 will be end of life by october 2025. Accessibility updates will not be deployed anymore to that Windows version.
Closing as works for me on Windows 11.
